### ISSUE TITLE
Implement a user database

### ISSUE THREAD

**Issue by natolh** [jeromecc/doctoctocbot#3]
Implementing a user database to store bot's contributors will allow :
- Exceed the 5.000 members limit of Twitter lists (Twitter lists are used to keep track of contributors).
- Allow to store more information about each user locally (currently the retweet script access only the contributors Twitter ID via a text file).

**Comment by jeromecc** [comment]
Hello!

The project does need a verified healthcare professionals database. The 5000 members limit of Twitter lists is not the main problem here, as 8 month after the creation of the bot, there are only 450 verified MDs in the main list and far less for other healthcare professionals.
The main objective would be to study characteristics (let's call them features) of verified MDs such as the content of their profile (bio, username), their social network (are they followed by verified MDs?), their previous usage of the hashtag or the answers they gave to a question tagged with doc(s)toctoc in order to add them to the whitelist automatically.
I am currently researching TensorFlow's Linear Model capabilities to implement a supervised learning algorithm to help human moderators. There are "only" 200.000 healthcare professionals in France and I don't think the bot will have to deal with more than a few thousands of them in the forseeable future so I don't think performance and big numbers will be an issue anytime soon. Twitter user's data is provided as json. See this discussion about the pros and cons of using a RDBMS or a native JSON DB to deal with Twitter's data: https://github.com/jimmoffitt/developer_advocate.blog/wiki/Part-1.-Introduction-to-Storing-Twitter-Data-in-a-Relational-Database.

I came to the conclusion that a native json database such as MongoDB and PyMongo library would be perfectly adapted to our challenges and much simpler to use because Twitter's API is changing frequently (so we would have to modify the schema of our SQL database frequently and the app could break with each change) and because we will never have to deal with millions of records so performance will never be a problem IMHO. This data will mostly be used to retrain the binary classifier algorithm (professional / not professional) every now and then and for research purpose, not for real time calculations for the app. What do you think @natolh?

**Comment by jeromecc** [comment]
As to the lists of user ids for the sake of RT/notRT I was thinking about a simple SQLite indexed column of INTs. Writing can be managed by one thread and reading by another. The only relevant info linked to each ID that I can think of now is the user id of the moderator who added it (i want to build a web app to allow multiple moderators to add new verified users) and maybe the timestamp.

**Comment by jeromecc** [comment]
After reading this http://www.sarahmei.com/blog/2013/11/11/why-you-should-never-use-mongodb/ I changed my mind about using a document database. I think that you were right. We should use a postgres database. PostgreSQL handles json very well if we want to store a few things in json or jsonb. Other advantage: interaction between the bot and the future Django website at https://doctoctoc.net will be easier with a single Postgres database.

**Comment by natolh** [comment]
@jeromecc yes exceeding the 5.000 limits of twitter list is not in the foresable future, but that was just one argument among the others. The bot need to access whitelisted user, if I'm not wrong ID are currently stored in a text file, it would be better in a database (it could be SQLite, MariaDB or PostgreSQL but that is a mere detail if we use an ORM such as SQLAlchemy). The 'store more information about the user' was refering to what is currently done by the 'getbio' and 'occurences' module.
